spec/controllers/tasks_controller_spec.rb

```diff
@@ -252,6 +278,28 @@
         expect { put_update }.to change { task.reload.license.name }.to('new_license')
       end
 
+      it 'updates the tasks labels' do
+        expect { put_update }.to change { task.reload.labels }.from([existing_label]).to([new_label])
+      end
+
+      context 'when requesting new label to be created' do
+        let(:changed_attributes) { {label_names: [not_existing_label_name]} }
+        let(:not_existing_label_name) { 'some new label' }
+
+        it 'creates new task label' do
+          expect { put_update }.to change(Label, :count).by(1)
```

Comment:
We should also add a test that reuses an existing label and ensure that the overall count isn't changed.

Reply:
I added a test for that.